airbyte-integrations/connectors/destination-firebolt/bootstrap.md

```diff
@@ -19,4 +19,6 @@ This connector uses [firebolt-sdk](https://pypi.org/project/firebolt-sdk/), whic
 ## Notes
 
 * Integration testing requires the user to have a running engine. Spinning up an engine can take a while so this ensures a faster iteration on the connector.
-* S3 is generally faster writing strategy and should be preferred.
\ No newline at end of file
+* S3 is generally faster writing strategy and should be preferred.
+* For normalization, quotes (") in json will be escaped \" and appear so in the final table. e.g. _airbyte_raw table contains `{"column": "my\"value\""}` final table would contain `my\"value\"` in column "column", therefore potentially diverging from original data.
```

Comment:
just for my understanding: where is this value coming from?

Reply:
This is when the source is sending some text data, that contains `"`. Those quotes will be escaped (as expected from JSON) but the resulting table would also have those quotes escaped, even after being normalised. This would be apparent when a sync from Postgres to Firebolt is done. Postgres would contain the above column with `my"value` in a cell, but normalised Firebolt table would contain `my\"value` instead.
I'll expand the doc to make it clearer.

Reply:
interesting. Is that because of how destination-firebolt works, or is that coming out of normalization? E.g. if in json_operations.sql, you replace the `json_extract_raw` with `json_extract`, would that fix this? I ran a test with destination-postgres locally, and didn't see this behavior:
```
edgao_test_replicated=> select * from _airbyte_raw_local_test_jsonl;
            _airbyte_ab_id            |       _airbyte_data        |  _airbyte_emitted_at
--------------------------------------+----------------------------+------------------------
 f3615098-f547-485e-99ae-2acdb70b4566 | {"x": {"y": "arst\"qwfp"}} | 2022-09-19 16:20:41+00
(1 row)

edgao_test_replicated=> select * from local_test_jsonl_x;
 _airbyte_local_test_jsonl_hashid |     y     |            _airbyte_ab_id            |  _airbyte_emitted_at   |    _airbyte_normalized_at     |        _airbyte_x_hashid
----------------------------------+-----------+--------------------------------------+------------------------+-------------------------------+----------------------------------
 bc69e12ea6ebc3367843bdeff27d64f2 | arst"qwfp | f3615098-f547-485e-99ae-2acdb70b4566 | 2022-09-19 16:20:41+00 | 2022-09-19 16:21:49.096006+00 | d87c91e9b843d6fe2eb8c9fc07ce5706
(1 row)
```

(I used a nested object, but non-nested fields should work the same way afaik)

Reply:
Yeah, the issue is with the SQL functions available in Firebolt specifically. Firebolt is different from Postgres since the latter has `jsonb_extract_path_text` which transforms strings with escaped " into their proper form. Firebolt has `json_extract(<field>, <path>, 'STRING')`, but unlike `jsonb_extract_path_text` which also works for non-string types (e.g. integers) `json_extract` will return null if the value is not string. 
```
-- firebolt
SELECT json_extract_raw('{"x":"abc\"d"}', 'x')  -- returns abc\"d 
SELECT json_extract('{"x":"abc\"d"}', 'x', 'STRING')  -- returns abc"d 
SELECT json_extract('{"x":1}', 'x', 'STRING')  -- returns Null

-- Postgres
SELECT jsonb_extract_path_text('{"c1":"ab\"c"}', 'c1') -- returns ab"c
SELECT jsonb_extract_path_text('{"c1":1}', 'c1'); -- returns 1
```